### PR TITLE
Cache jenkins client and make it lazy

### PR DESCRIPTION
**Pull request by frankh**
Without lazy=True, it polls for information on initialise

### CODE REVIEW

File: cabot/cabotapp/jenkins.py [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
+from jenkinsapi.jenkins import Jenkins
 
 logger = get_task_logger(__name__)
 
+JENKINS_CLIENT = None
 
-def _get_jenkins_client():
-    return Jenkins(settings.JENKINS_API, username=settings.JENKINS_USER, password=settings.JENKINS_PASS)
 
+def _get_jenkins_client(jenkins_config):
```
Comment (JeanFred) [inline cabot/cabotapp/jenkins.py]: The `jenkins_config` is not yet to be passed :)

Comment (frankh) [inline cabot/cabotapp/jenkins.py]: oops, fixed

File: cabot/cabotapp/jenkins.py [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
 
 logger = get_task_logger(__name__)
 
+JENKINS_CLIENT = None
 
-def _get_jenkins_client():
-    return Jenkins(settings.JENKINS_API, username=settings.JENKINS_USER, password=settings.JENKINS_PASS)
 
+def _get_jenkins_client():
+    global JENKINS_CLIENT
```
Comment (JeanFred) [inline cabot/cabotapp/jenkins.py]: Ah ah! That was what I was missing! :)

File: cabot/cabotapp/jenkins.py [hunk 3]
```diff
@@ -1,17 +1,24 @@
...
+    if JENKINS_CLIENT is None:
+        JENKINS_CLIENT = Jenkins(settings.JENKINS_API,
+                username=settings.JENKINS_USER,
```
Comment (JeanFred) [inline cabot/cabotapp/jenkins.py]: Indentation feels weird here ?

Comment (frankh) [inline cabot/cabotapp/jenkins.py]: yep, fixed

**Comment by JeanFred** [comment]
Testing locally: the `lazy` took a check run from 60s to 20s ; and caching the Jenkins client object down to 5s. Nice :)

**Pull request merged**